reasoner/resolution/answer/AnswerState.java

```diff
@@ -76,8 +77,8 @@ public static Initial of(ConceptMap conceptMap) {
             }
 
             public Optional<DownstreamVars.Unified> toDownstreamVars(Unifier unifier) {
-                Optional<ConceptMap> unified = unifier.unify(conceptMap());
-                return unified.map(conceptMap -> new DownstreamVars.Unified(this, conceptMap, unifier));
+                Optional<Pair<ConceptMap, Unifier.Requirements.Instance>> unified = unifier.unify(conceptMap());
+                return unified.map(unification -> new DownstreamVars.Unified(this, unification.first(), unifier, unification.second()));
```

Comment:
we can hide alllll the work to do with these extra requirements in the `AnswerState` ;) i think its pretty neat!